### PR TITLE
feature/search

### PR DESCRIPTION
**Pull request by hanhanW**
* Add feature for searching specified string or regexp in ELF.
* Fix issue #36 

---
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/41)

### CODE REVIEW

**Comment by david942j** [comment]
This pr also fixed https://github.com/peter50216/pwntools-ruby/issues/36

File: test/elf/elf_test.rb [hunk 1]
```diff
@@ -53,4 +53,21 @@ def test_address
     @elf.address = new_address
     assert_equal(old_main - old_address + new_address, @elf.symbols.main)
   end
+
+  def test_search
+    elf = ::Pwnlib::ELF::ELF.new(File.join(@data, 'lib32', 'libc.so.6'), checksec: false)
+    assert_equal([1, 0x15e613], elf.search('ELF').to_a)
```
Comment (david942j) [inline test/elf/elf_test.rb]: `[0x1, 0x15e613]`

**Comment by hanhanW** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/elf/elf_test.rb, line 59 at r1](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/41#-KsyTXZcOe3qYAvNqJk7-r1-59:-Kt0EiPwTdVGGsTs5Llq:b2nhexc) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/d8a97d666ae06c4fc5d6ba0d3904dc7dd92becf7/test/elf/elf_test.rb#L59)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

`[0x1, 0x15e613]`
</blockquote></details>

Fix

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/41)*

File: lib/pwnlib/elf/elf.rb [hunk 2]
```diff
@@ -130,6 +131,49 @@ def inspect
...
       end
 
+      # Yields the ELF's virtual address space for the specified string or regexp.
+      # Returns an Enumerator if no block given.
+      #
+      # @param [String, Regexp] needle
+      #   The specified string to search.
+      #
+      # @return [Enumerator<Integer>]
+      #   A enumerator for offsets in ELF's virtual address space.
```
Comment (david942j) [inline lib/pwnlib/elf/elf.rb]: An enumerator

**Comment by hanhanW** [comment]
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[lib/pwnlib/elf/elf.rb, line 141 at r2](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/41#-KsyTXZcOe3qYAvNqJk6-r2-141:-Kt4urelyw25j7gOrxq1:b2ni2nk) ([raw file](https://github.com/peter50216/pwntools-ruby/blob/61ae20ca32c60ad3e1b36402b0277520388c9d94/lib/pwnlib/elf/elf.rb#L141)):*
<details><summary><i>Previously, david942j wrote…</i></summary><blockquote>

An enumerator
</blockquote></details>

fix

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/41)*

**Comment by david942j** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/peter50216/pwntools-ruby/41#-:-KtBLduNJAg_2VpoWimm:bnfp4nl)*

**Comment by david942j** [comment]
👍

**Pull request merged**